contracts/Vault.sol

```diff
@@ -647,6 +649,8 @@ contract Vault is
             );
         }
 
+        _rebalanceBeforeTransfer(amount);
+
         underlying.safeTransfer(_to, amount);
```

Comment:
sounds like it would be a better pattern to have an internal function covering these two calls, instead of duplicating both of them, everywhere (I'm assuming they're always both together?)

wdyt?

Reply:
This is indeed correct. However, I'm not a fan of functions that imply multiple responsibilities in their name, but in this case, I don't have a strong opinion on that. I'm okay with leaving this as it is right now if that is acceptable to you as well.